api/src/main/java/com/microsoft/gctoolkit/GCToolKit.java

```diff
@@ -120,39 +123,41 @@ public void loadAggregationsFromServiceLoader() {
      * an Aggregation was registered, the {@code getAggregation} method will return
      * null if the Aggregation was not used in the analysis.
      *
-     * @param aggregationClass the Aggregation class to register.
+     * @param aggregation the Aggregation class to register.
      * @see Aggregation
      * @see JavaVirtualMachine
      */
-    public void registerAggregation(Class<? extends Aggregation> aggregationClass) {
-        registeredAggregations.add(aggregationClass);
+    public void registerAggregation(Aggregation aggregation) {
+        registeredAggregations.add(aggregation);
     }
 
     /**
      * Perform an analysis on a GC log file. The analysis will use the Aggregations
-     * that were {@link #registerAggregation(Class) registered}, if appropriate for
+     * that were {@link #registerAggregation(Aggregation) registered}, if appropriate for
      * the GC log file.
      *
-     * @param logFile The log to analyze, typically a
+     * @param dataSource The log to analyze, typically a
      *                   {@link SingleGCLogFile} or
      *                   {@link RotatingGCLogFile}.
      * @return a representation of the state of the Java Virtual Machine resulting
      * from the analysis of the GC log file.
      */
-    public JavaVirtualMachine analyze(GCLogFile logFile) throws IOException  {
-        //todo: revert this to DataSource to account for non-GC log data sources once we have a use case (maybe JFR but JFR timers drift badly ATM)
-        //setup message bus
-        //DataSourceBus dataSourceBus = setupDataSourceBus();
-        //JVMEventBus eventBus = setupJVMEventBus();
-        //registerParsers(dataSourceBus,logFile.diary());
-        JavaVirtualMachine javaVirtualMachine = loadJavaVirtualMachine(logFile);
-        //todo: do we need reflection????
+    public JavaVirtualMachine analyze(DataSource<?> dataSource) throws IOException  {
+        /*
+        Assembly....b
```

Comment:
Since the JavaVirtualMachine is in the API module, we shouldn't need to load it with the SPI. The comment is leftover from when reflection was used. I'll remove it.